### PR TITLE
Fix tests, shorten error messages, tweak foreclosure condition + ad back truffle config file

### PR DESCRIPTION
**Pull request by JasoonS**

### CODE REVIEW

File: contracts/WildcardSteward_v3.sol [hunk 1]
```diff
@@ -72,6 +65,14 @@ contract WildcardSteward_v3 is Initializable {
     mapping(address => uint256) public benefactorCredit;
     address public withdrawCheckerAdmin;
 
+    /*
+    31536000 seconds = 365 days
+
+    divisor = 365 days * 1000000000000
+            = 31536000000000000000
+    */
+    uint256 constant yearTimePatronagDenominator = 31536000000000000000;
```
Comment (JasoonS) [inline contracts/WildcardSteward_v3.sol:74]: I made this a global constant. The `constant` keyword ensures there is no extra gas cost.

File: contracts/WildcardSteward_v3.sol [hunk 2]
```diff
@@ -150,7 +151,7 @@ contract WildcardSteward_v3 is Initializable {
         require(
             !(deposit[tokenPatron] == 0 &&
                 totalPatronOwnedTokenCost[tokenPatron] > 0),
-            "You already own tokens but have no deposit"
+            "no deposit existing tokens"
```
Comment (JasoonS) [inline contracts/WildcardSteward_v3.sol:154]: I shortened most of the revert reason strings. They don't need to be verbose. I have went with the standard of all lower case (I've just seen that in quite a few other solidity repos, not sure why)

File: contracts/WildcardSteward_v3.sol [hunk 3]
```diff
@@ -165,14 +166,22 @@ contract WildcardSteward_v3 is Initializable {
...
         _;
     }
+    modifier notNullAddress(address checkAddress) {
+        require(checkAddress != address(0), "null address");
+        _;
+    }
+    modifier notSameAddress(address firstAddress, address secondAddress) {
+        require(firstAddress != secondAddress, "cannot be same address");
+        _;
+    }
```
Comment (JasoonS) [inline contracts/WildcardSteward_v3.sol:176]: These two checks were in two functions, so I thought it made sense to make them modifiers.

File: contracts/WildcardSteward_v3.sol [hunk 4]
```diff
@@ -551,7 +565,7 @@ contract WildcardSteward_v3 is Initializable {
 
             if (
                 patronageOwedByTokenPatron > 0 &&
-                patronageOwedByTokenPatron >= deposit[tokenPatron]
+                patronageOwedByTokenPatron > deposit[tokenPatron]
```
Comment (JasoonS) [inline contracts/WildcardSteward_v3.sol:568]: I mentioned this on slack :)

File: test/steward_benefactor_withdrawal.js [hunk 5]
```diff
@@ -292,7 +294,7 @@ contract("WildcardSteward Benefactor collection", (accounts) => {
         );
         assert.equal(
           benefactorFundsAfter.toString(),
-          totalDueBeforeForeclosure.toString(),
+          "0",
           "benefactor funds should stay zero"
```
Comment (JasoonS) [inline test/steward_benefactor_withdrawal.js:298]: This revert reason was actually incorrect before. Now it is correct ;)

File: contracts/WildcardSteward_v3.sol (hunk 1 above)
Comment (moose-code) [reply contracts/WildcardSteward_v3.sol:74]: excellent

File: test/steward_multi_patronage.js [hunk 6]
```diff
@@ -260,6 +260,7 @@ contract("WildcardSteward owed", (accounts) => {
     ]);
 
     if (!isCoverage)
+      // TODO: investigate why this test is flaky. I think it is the only flaky test in our whole test suite.
```
Comment (JasoonS) [inline test/steward_multi_patronage.js:263]: To my surprise, sometimes this test fails. Didn't investigate why yet

File: truffle-config.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
+const path = require("path");
```
Comment (JasoonS) [inline truffle-config.js:1]: I deleted this by mistake :see_no_evil: Anyway added it back. Made zero changes, no need to review.

File: contracts/WildcardSteward_v3.sol (hunk 2 above)
Comment (moose-code) [reply contracts/WildcardSteward_v3.sol:154]: Great, any chance we can have a comma between so it makes more sense?

no deposit, own tokens

File: contracts/WildcardSteward_v3.sol (hunk 4 above)
Comment (moose-code) [reply contracts/WildcardSteward_v3.sol:568]: Yeah think we are both onboard with this one.

**Pull request merged**